engine/src/types.rs

```diff
@@ -118,9 +118,9 @@ impl std::fmt::Display for ExpectedTypeList {
             _ => {
                 let mut iter = self.0.iter();
                 let first = iter.next().unwrap();
-                write!(f, "{{{}", first)?;
+                write!(f, "{{{first}")?;
                 for expected_type in iter {
-                    write!(f, ", {}", expected_type)?;
+                    write!(f, ", {expected_type}")?;
                 }
                 write!(f, "}}")
```

Comment:
nit: I might be missing something, but I think this whole block can be simplified?
```rust
write!(f, "{{")?;
for expected_type in self.0.iter() {
    write!(f, ", {expected_type}")?;
}
write!(f, "}}")
```

Reply:
But it means the first element would be prefixed by a comma?

Reply:
Ah, I missed the comma completely 😅